x/nft/keeper/grpc_query.go

```diff
@@ -97,29 +94,55 @@ func (k Keeper) NFTsOfClass(goCtx context.Context, r *nft.QueryNFTsOfClassReques
 		if err != nil {
 			return nil, err
 		}
-		return &nft.QueryNFTsOfClassResponse{
+		return &nft.QueryNFTsResponse{
 			Nfts:       nfts,
 			Pagination: pageRes,
 		}, nil
-	}
+	case len(r.ClassId) > 0 && len(r.Owner) == 0:
+		if err := nft.ValidateClassID(r.ClassId); err != nil {
+			return nil, err
+		}
 
-	nftStore := k.getNFTStore(ctx, r.ClassId)
-	pageRes, err := query.Paginate(nftStore, r.Pagination, func(_ []byte, value []byte) error {
-		var nft nft.NFT
-		if err := k.cdc.Unmarshal(value, &nft); err != nil {
-			return err
+		nftStore := k.getNFTStore(ctx, r.ClassId)
+		pageRes, err := query.Paginate(nftStore, r.Pagination, func(_ []byte, value []byte) error {
+			var nft nft.NFT
+			if err := k.cdc.Unmarshal(value, &nft); err != nil {
+				return err
+			}
+			nfts = append(nfts, &nft)
+			return nil
+		})
+
+		if err != nil {
+			return nil, err
+		}
+		return &nft.QueryNFTsResponse{
+			Nfts:       nfts,
+			Pagination: pageRes,
+		}, nil
+	case len(r.ClassId) == 0 && len(r.Owner) > 0:
+		owner, err := sdk.AccAddressFromBech32(r.Owner)
+		if err != nil {
+			return nil, err
 		}
-		nfts = append(nfts, &nft)
-		return nil
-	})
 
-	if err != nil {
-		return nil, err
+		pageRes, err := query.Paginate(k.prefixStoreNftOfClassByOwner(ctx, owner), r.Pagination, func(key []byte, value []byte) error {
+			classID, nftID := parseNftOfClassByOwnerStoreKey(key)
+			if n, has := k.GetNFT(ctx, classID, nftID); has {
+				nfts = append(nfts, &n)
+			}
+			return nil
+		})
+
+		if err != nil {
+			return nil, err
+		}
+		return &nft.QueryNFTsResponse{
+			Nfts:       nfts,
+			Pagination: pageRes,
+		}, nil
```

Comment:
lot of repeated code in the 3 cases above. Maybe we can put the error below (ErrINvalidRequest) in the `default` case, and after the switch put lines 137-143 (they are the same in all 3 cases).